File: src/main/java/toker/panel/entity/CraftingRequest.java
package toker.panel.entity;

import com.fasterxml.jackson.annotation.JsonView;

import javax.persistence.*;
import java.sql.Timestamp;

@Entity
@Table(name = "crafting_request")
public class CraftingRequest {

    private Integer id;
    private Timestamp creationTime;
    private Boolean taken;
    private CraftingRecipe craftingRecipe;
    private Player player;
    private CraftingStationInstance craftingStationInstance;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    @Column(name = "creation_time", nullable = false)
    public Timestamp getCreationTime() {
        return creationTime;
    }

    public void setCreationTime(Timestamp creationTime) {
        this.creationTime = creationTime;
    }

    @Column(name = "is_taken", nullable = false)
    public Boolean isTaken() {
        return taken;
    }

    public void setTaken(Boolean taken) {
        this.taken = taken;
    }

    @ManyToOne
    @JoinColumn(name = "recipe_id", referencedColumnName = "id", nullable = false)
    @JsonView(View.CraftingRecipe.class)
    public CraftingRecipe getCraftingRecipe() {
        return craftingRecipe;
    }

    public void setCraftingRecipe(CraftingRecipe craftingRecipe) {
        this.craftingRecipe = craftingRecipe;
    }

    @ManyToOne
    @JoinColumn(name = "player_id", referencedColumnName = "id", nullable = false)
    @JsonView(View.Player.class)
    public Player getPlayer() {
        return player;
    }

    public void setPlayer(Player player) {
        this.player = player;
    }

    @ManyToOne
    @JoinColumn(name = "station_instance_id", referencedColumnName = "id", nullable = false)
    @JsonView(View.CraftingStationInstance.class)
    public CraftingStationInstance getCraftingStationInstance() {
        return craftingStationInstance;
    }

    public void setCraftingStationInstance(CraftingStationInstance craftingStationInstance) {
        this.craftingStationInstance = craftingStationInstance;
    }

    public interface View {
        interface CraftingStationInstance {}
        interface Player {}
        interface CraftingRecipe {}
    }
}

File: src/main/java/toker/panel/entity/CraftingRequest.kt
package toker.panel.entity

import com.fasterxml.jackson.annotation.JsonView
import java.sql.Timestamp
import javax.persistence.*

@Entity
@Table(name = "crafting_request")
class CraftingRequest(
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        @Column(name = "id", nullable = false)
        var id: Int? = null,

        @Column(name = "creation_time", nullable = false)
        var creationTime: Timestamp? = null,

        @Column(name = "is_taken", nullable = false)
        var isTaken: Boolean? = null,

        @ManyToOne
        @JoinColumn(name = "recipe_id", referencedColumnName = "id", nullable = false)
        @JsonView(View.CraftingRecipe::class)
        var craftingRecipe: CraftingRecipe? = null,

        @ManyToOne
        @JoinColumn(name = "player_id", referencedColumnName = "id", nullable = false)
        @JsonView(View.Player::class)
        var player: Player? = null,

        @ManyToOne
        @JoinColumn(name = "station_instance_id", referencedColumnName = "id", nullable = false)
        @JsonView(View.CraftingStationInstance::class)
        var craftingStationInstance: CraftingStationInstance? = null
) {
    interface View {
        interface CraftingStationInstance
        interface Player
        interface CraftingRecipe
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        other as CraftingRequest
        if (id != other.id) return false
        return true
    }

    override fun hashCode(): Int {
        return id ?: 0
    }
}